test/Extensions.Mcp.EndToEndTests/ProtocolTests/Default/DefaultInitializationTests.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Can we break out this PR? One PR to introduce the test framework, and follow up PRs to add E2E tests